api/api.yaml

```diff
@@ -0,0 +1,619 @@
+  /verifier/api/v1/startsiop:
+    get:
+      tags:
+        - verifier
+        - api 
+      parameters:
+        - $ref: '#/components/parameters/QueryState'
+      operationId: VerifierAPIStartSIOP
+      summary: Initiates the siop flow and returns the 'openid://...' connection string
+      responses:
+        '200': 
+          description: The connection string 
+          content:
+            text/plain:
+              schema:
+                type: string
+                example: openid://?scope=dsba.credentials.presentation.PacketDeliveryService&response_type=vp_token&response_mode=post&client_id=did:key:z6MktZy7CErCqdLvknH6g9YNVpWupuBNBNovsBrj4DFGn4R1&redirect_uri=http://localhost:3000/verifier/api/v1/authenticationresponse&state=&nonce=BfEte4DFdlmdO7a_fBiXTw
+  /verifier/api/v1/authenticationrequest:
+    $ref: '#/paths/~1verifier~1api~1v1~1startsiop'
+  /verifier/api/v1/authenticationresponse:
+    post:
+      tags:
+        - verifier
+        - api 
+      parameters:
+        - $ref: '#/components/parameters/QueryState'
+      operationId: VerifierAPIAuthenticationResponse
+      summary: Stores the credential for the given session
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/VCContainer'
+      responses:
+        '200':  
+          description: Ok when it worked
+          content:
+            text/plain:
+              schema:
+                type: string
+                enum: 
+                  - "ok"
+  /wallet/api/v1/selectcredential:
+    get:
+      tags:
+        - wallet
+        - frontend
+      operationId: WalletPageSelectCredential
+      summary: Present a rendered page with the available credentials
+      responses:
+        '200':
+          description: The rendered page
+          content:
+            text/html:
+              schema:
+                type: object
+  /wallet/api/v1/sendcredential:
+    get:
+      tags:
+        - wallet
+        - frontend
+      parameters:
+        - $ref: '#/components/parameters/QueryState'
+        - $ref: '#/components/parameters/QueryId'
+        - $ref: '#/components/parameters/RedirectUri'
+      operationId: WalletPageSendCredential
+      summary: Executes a post with the given credential to the redirect_uri and returns a rendered page, presenting the post-result
+      responses:
+        '200':
+          description: The rendered page
+          content:
+            text/html:
+              schema:
+                type: object
+  /core/api/v1/createdid:
+    get:
+      tags:
+        - core
+        - api
+      operationId: CoreAPICreateDID
+      summary: Wrapper for the waltid-ssikit custodian endpoint for creating did's of type key
+      responses:
+        '200':
+          description: Creation result from the ssikit
+          content:
+            application/json:
```

Comment:
Why is this application/json? The method responds a plain string, this could lead to clients failing the response, since the content does not match the type.

Reply:
Yes, either the string should be quoted (and then is a valid JSON string), or the mime type changed.
In the VC WG there were endless discussions about mime types and the sex of angels, and they are not yet finalized. For consistency with the "importance" of a DID, probably the right approach is to use application/json and quote the string.
But I do not care.

Also, the "proper" mime type for DID Documents is application/did+ld+json, but I think it still has not been accepted by IANA (because the two suffixes). The other option is application/did+json, which is the one being mentioned in the documents. But I used application/json because is is also valid and for me clearer.

So, you decide on the mime type of the DID as standalone.